### PR TITLE
INT: Implement "FlattenUseStatementsIntention"

### LINKED ISSUE #2577: Intention to nest/flatten imports
```rust
use std::error::Error;
use std::io::Write;
use std::path::PathBuf;
```

<-->

```rust
use std::{
    error::Error,
    io::Write,
    path::PathBuf,
};
```

### PR DESCRIPTION
**Pull request by mibac138**
I believe this closes #2577.
Please feel free to give me your feedback

### CODE REVIEW

File: src/test/kotlin/org/rust/ide/intentions/FlattenUseStatementsIntentionTest.kt [hunk 1]
```diff
@@ -0,0 +1,265 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.intentions
+
+class FlattenUseStatementsIntentionTest : RsIntentionTestBase(FlattenUseStatementsIntention()) {
```
Comment (mchernyavsky) [inline src/test/kotlin/org/rust/ide/intentions/FlattenUseStatementsIntentionTest.kt]: Let's add test for use item with alias:
```rust
use std::collections::{HashMap as HM, HashSet as HS};

```

File: src/main/kotlin/org/rust/ide/intentions/FlattenUseStatementsIntention.kt [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+        val useSpecks: List<RsUseSpeck>
+        val root: PsiElement
+        val firstOldElement: PsiElement
+        fun createElements(paths: List<String>, project: Project): List<PsiElement>
+        val oldElements: List<PsiElement>
+        val cursorOffset: Int
+        val basePath: String
+    }
+
+    override fun findApplicableContext(project: Project, editor: Editor, element: PsiElement): Context? {
```
Comment (mchernyavsky) [inline src/main/kotlin/org/rust/ide/intentions/FlattenUseStatementsIntention.kt]: This intention should take into account attributes and visibility:

```rust
#[cfg(foo)]
pub use std::collections::{/*caret*/HashMap, HashSet};
```

```rust
use std::collections::HashMap;
use std::collections::HashSet;
```

Comment (mibac138) [reply src/main/kotlin/org/rust/ide/intentions/FlattenUseStatementsIntention.kt]: Should this result in
```rust
#[cfg(foo)]
pub use std::collections::HashMap;
#[cfg(foo)]
pub use std::collections::HashSet;
```
I am specifically asking about the attribute

**Comment by mibac138** [comment]
@mchernyavsky I have applied all of your comments

**Comment by mchernyavsky** [comment]
Thanks a lot!
bors r+

**Pull request merged**